src/helpers/appFunctions.ts

```diff
@@ -0,0 +1,229 @@
+
+function addToCart(elem: Node): void {
+    const itemID: number = parseInt((elem as HTMLButtonElement).value);
+    const clickItem: IPrototypeItem = <IPrototypeItem>shoes.find((el) => el.id === itemID);
+    const cartItem: ItemCart = new ItemCart(
+        clickItem.id,
+        clickItem.name,
+        clickItem.brand,
+        clickItem.category,
+        clickItem.thumbnail,
+        1,
+        clickItem.stock,
+        clickItem.price
+    );
+    if (arrCart.length === 0) {
+        arrCart.push(cartItem);
+    } else {
+        const findElem: ItemCart | undefined = arrCart.find((el) => el.id === itemID);
+        if (findElem !== undefined) {
+            findElem.addAmount();
+        } else {
+            arrCart.push(cartItem);
+        }
+    }
+    updateHeader();
+}
+
+function cartButtonAddClick(): void {
+    const itemsAddCartButton: NodeList = <NodeList>document.querySelectorAll('.btn-to-cart');
+
+    itemsAddCartButton.forEach((el) => {
+        el.addEventListener('click', () => addToCart(el));
+    });
+}
+
+//Remove Item from Cart
+function removeItemFromCart(id: number): void {
+    arrCart.splice(id, 1);
+    updatePaginParam();
+    updateHeader();
+    App.renderNewPage('cart');
+}
+
+function clearCart(): void {
+    arrCart.splice(0, arrCart.length);
+    setPerPage(3);
+    saveLocalStorage();
+    removeSearchParams(['category']);
+}
+
+function shoesImportToItemCart(shoe: IPrototypeItem): ItemCart {
+    const newItemCart: ItemCart = new ItemCart(
+        shoe.id,
+        shoe.name,
+        shoe.brand,
+        shoe.category,
+        shoe.thumbnail,
+        1,
+        shoe.stock,
+        shoe.price
+    );
+    return newItemCart;
+}
+
+//Add / Remove Item Cart
+function addItemToCart(itemID: number): void {
+    if (findInCart(itemID) < 0) {
+        arrCart.push(shoesImportToItemCart(shoes[itemID - 1]));
+    } else {
+        arrCart.splice(findInCart(itemID), 1);
+    }
+    updateHeader();
+}
+
+//Buy Now
+function buyNow(itemID: number): void {
+    if (findInCart(itemID) < 0) arrCart.push(shoesImportToItemCart(shoes[itemID - 1]));
+    updateHeader();
+    App.renderNewPage(PageIDs.CartPage, true);
+    new Form().listen();
+}
+
+//loading parameter from localStorage
+function loadLocalStorage() {
+    if (localStorage.getItem('arrCart')) {
+        arrCart = JSON.parse(String(localStorage.getItem('arrCart'))).map((el: ItemCart) => {
+            const { id, name, brand, category, thumbnail, amount, limit, price } = el;
+            return new ItemCart(id, name, brand, category, thumbnail, amount, limit, price);
+        });
+        updateHeader();
+        loadIconForItems();
+    } else {
+        arrCart = [];
+    }
+    if (localStorage.getItem('perPage')) {
+        setPerPage(parseInt(String(localStorage.getItem('perPage'))));
+    } else {
+        setPerPage(3);
+    }
+    if (localStorage.getItem('currPage')) {
+        setCurrPage(parseInt(String(localStorage.getItem('currPage'))));
```

Comment:
also if you see that your code repeat some times, you should think about creating function. for example:
 const parseLocalStorage = (string) => {
 return parseInt(String(localStorage.getItem(string)))
}
than: 
localStorage.getItem('perPage') ?  setPerPage(parseLocalStorage('perPage')) : setPerPage(3);
don't change your code here if want